cli/azd/pkg/project/framework_service.go

```diff
@@ -56,6 +65,11 @@ type FrameworkService interface {
 	// This is useful if the framework needs to subscribe to any service events
 	Initialize(ctx context.Context, serviceConfig *ServiceConfig) error
 
+	// Gets the requirements for the language or framework service.
+	// This enables more fine grain control on whether the language / framework
+	// supports or requires lifecycle commands such as restore, build, and package
```

Comment:
"supports or requires" may be confusing. It seems like from the way we've written it, it means the exact requirements for a particular lifecycle.